plugin-server/src/worker/vm/promise-manager.ts

```diff
@@ -0,0 +1,27 @@
+import { PluginsServerConfig } from '../../types'
+
+export class PromiseManager {
+    pendingPromises: Set<Promise<any>>
+    config: PluginsServerConfig
+
+    constructor(config: PluginsServerConfig) {
+        this.pendingPromises = new Set()
+        this.config = config
+    }
+
+    public trackPromise(promise: Promise<any>): Promise<any> {
+        this.pendingPromises.add(promise)
+
+        promise.finally(() => {
+            this.pendingPromises.delete(promise)
+        })
+
+        return promise
+    }
+
+    public async awaitPromisesIfNeeded() {
+        while (this.pendingPromises.size > this.config.MAX_PENDING_PROMISES_PER_WORKER) {
```

Comment:
TS warning...

Reply:
Wait, where? I don't get this and code quality didn't catch it either

Reply:
[Here](https://github.com/PostHog/posthog/pull/8573/files):

![image](https://user-images.githubusercontent.com/53387/154239053-2e8e5381-91e6-43b6-ac6e-0140a035b7f1.png)

There seem to be a lot of them from the plugin server showing up at the bottom of PR reviews lately.

![image](https://user-images.githubusercontent.com/53387/154239139-44eefb73-3ac8-4ceb-a05b-6895e25378f2.png)

![image](https://user-images.githubusercontent.com/53387/154239122-a232bd82-8176-4ff2-9960-1cbd6e5f8994.png)

Reply:
ah right my IDE is not showing me warnings, will update